src/arrayMethodJoin.js

```diff
@@ -4,8 +4,31 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (Array.isArray(this) && this.length > 1) {
+      let str;
+
+      if (this[0] === null || this[0] === undefined) {
```

Comment:
No need to duplicate this check. You can check this element in a loop too, and make str = '' initially.